middleware/kubernetes/kubernetes.go

```diff
@@ -270,32 +314,66 @@ func (k *Kubernetes) getRecordsForServiceItems(serviceItems []*api.Service, zone
 }
 
 // Get performs the call to the Kubernetes http API.
-func (k *Kubernetes) Get(namespace string, nsWildcard bool, servicename string, serviceWildcard bool, typeName string) ([]*api.Service, error) {
+func (k *Kubernetes) Get(namespace string, nsWildcard bool, servicename string, serviceWildcard bool, endpointname string, port string, portWildcard bool, protocol string, protocolWildcard bool, typeName string) (services []service, err error) {
 	switch {
 	case typeName == "pod":
-		return nil, fmt.Errorf("pod not implemented")
+		return nil, fmt.Errorf("%v not implemented", typeName)
 	default:
-		return k.getServices(namespace, nsWildcard, servicename, serviceWildcard)
+		return k.getServices(namespace, nsWildcard, servicename, serviceWildcard, endpointname, port, portWildcard, protocol, protocolWildcard)
 	}
 }
 
-func (k *Kubernetes) getServices(namespace string, nsWildcard bool, servicename string, serviceWildcard bool) ([]*api.Service, error) {
+func (k *Kubernetes) getServices(namespace string, nsWildcard bool, servicename string, serviceWildcard bool, endpointname string, port string, portWildcard bool, protocol string, protocolWildcard bool) ([]service, error) {
```

Comment:
IMO, The wildcard params are clunky, and not really necessary to pass along through the cumbersome chain of private functions.  They can be computed where needed (at the end of the chain in getServices()), and still serve the purpose (which I presume was the efficiency of not having to check for wildcards at every comparison).